src/senselab/utils/data_structures/dataset.py

```diff
@@ -0,0 +1,90 @@
+"""Data structures relevant for managing datasets."""
+
+import uuid
+from typing import Any, Dict, List
+
+from pydantic import BaseModel, Field, field_validator
+
+
+class Participant(BaseModel):
+    """Data structure for a participant in a dataset."""
+
+    id: str = Field(default_factory=lambda: str(uuid.uuid4()))
+    metadata: Dict = Field(default={})
+
+    @field_validator("id", mode="before")
+    def set_id(cls, v: str) -> str:
```

Comment:
I don't know if this will ever get run because you have a default_factory if the id is not provided so probably not needed.